Source/JavaScriptCore/dfg/DFGAbstractInterpreterInlines.h

```diff
@@ -1520,6 +1520,26 @@ bool AbstractInterpreter<AbstractStateType>::executeEffects(unsigned clobberLimi
     }
 
     case ToLowerCase: {
+        AbstractValue& property = forNode(m_graph.child(node, 0));
+        if (JSValue value = property.value()) {
+            if (value.isString()) {
+                JSString* string = asString(value);
+                if (const StringImpl* a = asString(string)->tryGetValueImpl()) {
+                    bool lower = true;
+                    for (unsigned index = 0; index < a->length(); ++index) {
+                        UChar character = a->at(index);
+                        if (!isASCII(character) || isASCIIUpper(character)) {
+                            lower = false;
+                            break;
+                        }
+                    }
+                    if (lower) {
```

Comment:
Maybe I'm missing something, but in the case where it isn't lower case already, couldn't you lower it here and setConstant? Either way the change looks good.

Reply:
Yeah, we can do that, but for now, I've just converted it only when the condition meets. We can later extend it by using LazyJSConstant node.